### ISSUE TITLE
Prettier not honoring plugins

### ISSUE THREAD

**Issue by affanshahid** [zed-industries/zed#8841]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Using Zed with a project that uses Prettier with some plugins. It seems to ignore the plugins when formatting files.
The prettier config:
```
{
  "singleQuote": true,
  "trailingComma": "all",
  "plugins": ["prettier-plugin-organize-imports"]
}
```

Prettier LSP logs:
```
stderr: Resolved config: {"singleQuote":true,"trailingComma":"all","plugins":["prettier-plugin-organize-imports"]}, will format file '/Users/affan/Development/fhs-website/app/(portal)/resident/payments/page.tsx' with options: {"singleQuote":true,"trailingComma":"all","plugins":[],"parser":"typescript","path":"/Users/affan/Development/fhs-website/app/(portal)/resident/payments/page.tsx"}
stderr: Resolved config: {"singleQuote":true,"trailingComma":"all","plugins":["prettier-plugin-organize-imports"]}, will format file '/Users/affan/Development/fhs-website/app/(portal)/resident/payments/page.tsx' with options: {"singleQuote":true,"trailingComma":"all","plugins":[],"parser":"typescript","path":"/Users/affan/Development/fhs-website/app/(portal)/resident/payments/page.tsx"}
stderr: Resolved config: {"singleQuote":true,"trailingComma":"all","plugins":["prettier-plugin-organize-imports"]}, will format file '/Users/affan/Development/fhs-website/app/(portal)/resident/payments/page.tsx' with options: {"singleQuote":true,"trailingComma":"all","plugins":[],"parser":"typescript","path":"/Users/affan/Development/fhs-website/app/(portal)/resident/payments/page.tsx"}
```

### Environment

Zed: v0.125.1 (Zed Preview)
OS: macOS 14.2.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

[Zed.log](https://github.com/zed-industries/zed/files/14485092/Zed.log)

**Comment by RemcoSmitsDev** [comment]
@mrnugget

**Comment by mrnugget** [comment]
I'm 99% sure this was fixed by https://github.com/zed-industries/zed/commit/9c6a0d98eda255ec934cff1855f5ad1353eb0c86 which ended up in [`0.126.0`](https://zed.dev/releases/preview/0.126.0). Can you try that version and see if the problem still shows up?

**Comment by timcole** [comment]
Hey @mrnugget, I'm on Zed Preview 0.126.1 and `@trivago/prettier-plugin-sort-imports` plugin still doesn't work.

```
stderr: Resolved config: {"semi":true,"trailingComma":"all","singleQuote":true,"printWidth":80,"tabWidth":2,"importOrder":["^@type/(.*)$","^@generated/(.*)$","^@lib/(.*)$","^react-use/(.*)$","^@hooks/(.*)$","^@components/(.*)$","^@email/(.*)$","^@(server|trpc)/(.*)$","^[./]","^simple-icons$","^@/icons/(.*)$"],"importOrderSeparation":true,"plugins":["@trivago/prettier-plugin-sort-imports"]}, will format file '/Users/tim/projects/socialblade/socialblade.com/server/routers/admin/index.ts' with options: {"semi":true,"trailingComma":"all","singleQuote":true,"printWidth":80,"tabWidth":2,"importOrder":["^@type/(.*)$","^@generated/(.*)$","^@lib/(.*)$","^react-use/(.*)$","^@hooks/(.*)$","^@components/(.*)$","^@email/(.*)$","^@(server|trpc)/(.*)$","^[./]","^simple-icons$","^@/icons/(.*)$"],"importOrderSeparation":true,"plugins":[],"parser":"typescript","path":"/Users/tim/projects/socialblade/socialblade.com/server/routers/admin/index.ts"}
```

**Comment by RemcoSmitsDev** [comment]
@timcole I have created a pull request to fix this issue, so sit tight😀.

**Comment by mrnugget** [comment]
This was fixed by #9193 which will go out in the next preview.
